src/main/java/com/minecolonies/coremod/colony/buildings/views/AbstractBuildingView.java

```diff
@@ -153,13 +155,22 @@ public boolean isRepairing()
 
     /**
      * Open the associated BlockOut window for this building.
+     * If the player is sneaking open the inventory else open the GUI directly.
+     * @param isPlayerSneaking if the player is sneaking.
      */
-    public void openGui()
+    public void openGui(final boolean isPlayerSneaking)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Provide multiple methods instead of using "isPlayerSneaking" to determine which action to take. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2301)